From 3e467c4182cb5fa47c326e176bce00661dcfd580 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 3 Oct 2025 17:03:54 +0200
Subject: [PATCH] ARM: dts: stm32: add phase tags marking on stm32mp13

bootph-all/bootph-some-ram as phase tags was added to dt-schema
(dtschema/schemas/bootph.yaml) to describe various node usage during
boot phases with DT.

Describe them for STM32MP13 boot devices.

Change-Id: Ied02b70e65184378960b5ae26ac1de7801e301cc
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/496294
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 22 ++++++++++++++++++++++
 arch/arm/boot/dts/st/stm32mp135.dtsi |  1 +
 2 files changed, 23 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp131.dtsi b/arch/arm/boot/dts/st/stm32mp131.dtsi
index 50f3e02ba508..99934b8e6ff4 100644
--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -42,6 +42,7 @@ arm_wdt: watchdog {
 
 	firmware {
 		optee: optee {
+			bootph-all;
 			method = "smc";
 			compatible = "linaro,optee-tz";
 			interrupt-parent = <&intc>;
@@ -51,6 +52,7 @@ optee: optee {
 		};
 
 		scmi: scmi {
+			bootph-all;
 			compatible = "linaro,scmi-optee";
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -62,11 +64,13 @@ scmi_perf: protocol@13 {
 			};
 
 			scmi_clk: protocol@14 {
+				bootph-all;
 				reg = <0x14>;
 				#clock-cells = <1>;
 			};
 
 			scmi_reset: protocol@16 {
+				bootph-all;
 				reg = <0x16>;
 				#reset-cells = <1>;
 			};
@@ -104,6 +108,7 @@ intc: interrupt-controller@a0021000 {
 	};
 
 	psci {
+		bootph-some-ram;
 		compatible = "arm,psci-1.0";
 		method = "smc";
 	};
@@ -155,6 +160,7 @@ cpu_crit: cpu-crit0 {
 	};
 
 	soc {
+		bootph-all;
 		compatible = "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -841,6 +847,7 @@ dmamux1: dma-router@48002000 {
 		};
 
 		rcc: rcc@50000000 {
+			bootph-all;
 			compatible = "st,stm32mp13-rcc", "syscon";
 			reg = <0x50000000 0x1000>;
 			#clock-cells = <1>;
@@ -854,6 +861,7 @@ rcc: rcc@50000000 {
 		};
 
 		pinctrl: pinctrl@50002000 {
+			bootph-all;
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "st,stm32mp135-pinctrl";
@@ -861,6 +869,7 @@ pinctrl: pinctrl@50002000 {
 			interrupt-parent = <&exti>;
 
 			gpioa: gpio@50002000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -873,6 +882,7 @@ gpioa: gpio@50002000 {
 			};
 
 			gpiob: gpio@50003000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -885,6 +895,7 @@ gpiob: gpio@50003000 {
 			};
 
 			gpioc: gpio@50004000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -897,6 +908,7 @@ gpioc: gpio@50004000 {
 			};
 
 			gpiod: gpio@50005000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -909,6 +921,7 @@ gpiod: gpio@50005000 {
 			};
 
 			gpioe: gpio@50006000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -921,6 +934,7 @@ gpioe: gpio@50006000 {
 			};
 
 			gpiof: gpio@50007000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -933,6 +947,7 @@ gpiof: gpio@50007000 {
 			};
 
 			gpiog: gpio@50008000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -945,6 +960,7 @@ gpiog: gpio@50008000 {
 			};
 
 			gpioh: gpio@50009000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -957,6 +973,7 @@ gpioh: gpio@50009000 {
 			};
 
 			gpioi: gpio@5000a000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -1049,6 +1066,7 @@ exti: interrupt-controller@5000d000 {
 		};
 
 		syscfg: syscon@50020000 {
+			bootph-all;
 			compatible = "st,stm32mp157-syscfg", "syscon";
 			reg = <0x50020000 0x400>;
 			clocks = <&rcc SYSCFG>;
@@ -1156,6 +1174,7 @@ usbh_ehci: usb@5800d000 {
 		};
 
 		iwdg2: watchdog@5a002000 {
+			bootph-all;
 			compatible = "st,stm32mp1-iwdg";
 			reg = <0x5a002000 0x400>;
 			interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
@@ -1192,6 +1211,7 @@ rtc: rtc@5c004000 {
 		};
 
 		bsec: efuse@5c005000 {
+			bootph-all;
 			compatible = "st,stm32mp13-bsec";
 			reg = <0x5c005000 0x400>;
 			#address-cells = <1>;
@@ -1219,6 +1239,7 @@ ethernet_mac2_address: mac2@ea {
 		};
 
 		etzpc: bus@5c007000 {
+			bootph-all;
 			compatible = "st,stm32-etzpc", "simple-bus";
 			reg = <0x5c007000 0x400>;
 			#address-cells = <1>;
@@ -1816,6 +1837,7 @@ stmmac_axi_config_1: stmmac-axi-config {
 			};
 
 			usbphyc: usbphyc@5a006000 {
+				bootph-all;
 				#address-cells = <1>;
 				#size-cells = <0>;
 				#clock-cells = <0>;
diff --git a/arch/arm/boot/dts/st/stm32mp135.dtsi b/arch/arm/boot/dts/st/stm32mp135.dtsi
index d95b3d6d947f..850929fd9bc0 100644
--- a/arch/arm/boot/dts/st/stm32mp135.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp135.dtsi
@@ -19,6 +19,7 @@ dcmipp: dcmipp@5a000000 {
 	};
 
 	ltdc: display-controller@5a001000 {
+		bootph-some-ram;
 		compatible = "st,stm32-ltdc";
 		reg = <0x5a001000 0x400>;
 		interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>,
-- 
2.53.0

